docs/docs/how-to/adding-common-features/adding-forms.md

```diff
@@ -5,179 +5,95 @@ examples:
     href: "https://github.com/gatsbyjs/gatsby-functions-beta/tree/main/examples/basic-form"
 ---
 
-Gatsby is built on top of React. So anything that is possible with a React form is possible in Gatsby. Additional details about how to create React forms can be found in the [React forms documentation](https://reactjs.org/docs/forms.html) (which happens to be built with Gatsby!)
+Forms are common to capture user information from site visitors. They range in complexity from simple newsletter email capture boxes, to detailed lead capture forms and signup pages requiring visitors to enter their first & last name, email, and several additional self-identificaton questions.
 
-Start with the following page.
+The humble `<form>` element has evolved a lot since it was created in 1995. By default, a `<form>` stores its input internally and refresh the page when you `submit`. Today, developers generally want to store input in Javascript so you can validate fields and send the data to a third-party system.
 
-```jsx:title=src/pages/index.js
-import React from "react"
+Website forms have several "concerns" like data entry storage, validating fields, form submission, form storage, and other event triggers. This documentation will walk through each concern.
 
-export default function Home() {
-  return <div>Hello world!</div>
-}
-```
+Forms in Gatsby are built on top of React, so in a couple places we'll be linking to the React docs for code examples. In addition, the React ecosystem also offers some popular, well-maintained libraries like [Formik](https://github.com/jaredpalmer/formik) or [Final Form](https://github.com/final-form/react-final-form) to speed up your development process.
 
-This Gatsby page is a React component. When you want to create a form, you need to store the state of the form - what the user has entered. Convert your function (stateless) component to a class (stateful) component.
+## Form concerns
 
-```jsx:title=src/pages/index.js
-import React from "react"
+### Client-side data entry
 
-export default class IndexPage extends React.Component {
-  render() {
-    return <div>Hello world!</div>
+When users type into the input, you want to persist their input, as well as store it in a JavaScript field that will later be available to your form validation and submission logic.
```

Comment:
what do you mean by "persist their input"?

Reply:
great question 🤔